redundant-bmc/src/persistent_data.cpp

```diff
@@ -63,4 +85,68 @@ void remove(std::string_view name, const std::filesystem::path& path)
     }
 }
 
+void logFailover(const std::filesystem::path& dirPath, size_t bmcPos,
+                 Requester requester, const time_t& timestamp)
+{
+    try
+    {
+        auto file = util::makeFailoverLogFilePath(dirPath, bmcPos);
+        auto logs = util::readFile(file);
+
+        if (!logs.has_value())
+        {
+            logs = nlohmann::json::array();
+        }
+
+        if (!logs->is_array())
+        {
+            lg2::error("failover log {FILE} isn't a JSON array", "FILE", file);
+            return;
+        }
+
+        // If at the max number of entries, drop the oldest.
+        if (logs->size() == maxFailoverLogEntries)
+        {
+            logs->erase(logs->begin());
+        }
+
+        // Just save the last segment of the enum string, e.g. 'Host'.
+        auto r = sdbusplus::common::xyz::openbmc_project::control::Failover::
+            convertRequesterToString(requester);
+        r = r.substr(r.find_last_of('.') + 1);
```

Comment:
ok, string stuff always makes my head hurt. Why would this not be a "- 1" to go back one from the "."?